src/audio/kpb.c

```diff
@@ -615,8 +616,13 @@ static int kpb_copy(struct comp_dev *dev)
 							 "failed.");
 				goto out;
 			}
-			if (kpb->buffered_data < kpb->buffer_size)
-				kpb->buffered_data += copy_bytes;
+			if (kpb->buffered_data < kpb->buffer_size) {
+				hb_free_space = kpb->buffer_size -
+					kpb->buffered_data;
+
+				kpb->buffered_data +=
```

Comment:
Why do you think buffered_data should not exceed history buffer size? It is OK to exceed it, the only problem here is a lack of check if we don't overflow size_t max value. I would add a condition that once we hit max limit we no longer update and that would be enough IMHO. Don't you think?